Reactive Colours and Modal tweaks

aam-git opened the pull request:
Updates in this pull request

- colours change reactively when colors variable changes from outside of the plugin, this allows you to use sessions to sync colour pallets across multiple editors and dynamically add colours as they are used
- added native color picker to the modal input by switching out the input type depending on if its a color picker needed (detected by event type)
- only load the modal html when its needed not on runtime to facilitate the above and save extra html load when may not be needed, rather than the current show/hide type.

![image](https://user-images.githubusercontent.com/22966791/144461461-e9103eab-db1a-414b-afdb-b2c381b02bbf.png)

fuzzthink reviewed (commented):
This reads better imo. EDIT: Wouldn't need block too. @aam-git 

```js
  $: btns = colors
      .map((color) => ({ color }))
      .concat([{ text: '#', modal: true }]);
```

aam-git commented:
> This reads better imo. EDIT: Wouldn't need block too. @aam-git
> 
> ```js
>   $: btns = colors
>       .map((color) => ({ color }))
>       .concat([{ text: '#', modal: true }]);
> ```

Nice, just updated the pull request to add this :)

EDIT: updated to remove blocks

fuzzthink reviewed (approved):
lgtm. @nenadpnc Just noticed spaces are not consistent in the codebase. There is a mix of  tabs, 2 spaces, and 4 spaces. After this is approved and merged, I can submit a PR to convert all to the js standard of 2 spaces.

aam-git commented:
> lgtm. @nenadpnc Just noticed spaces are not consistent in the codebase. There is a mix of tabs, 2 spaces, and 4 spaces. After this is approved and merged, I can submit a PR to convert all to the js standard of 2 spaces.
> 
> EDIT: @aam-git If you change the indentation back to 2 spaces, it will de-noise the commit histories.

Done...  Odd, not sure why vscode did that, I don't normally let it format svelte files.

nenadpnc reviewed (approved):
🥇
